src/ucp/core/ucp_mm.c

```diff
@@ -94,6 +97,8 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
         md_attr = &context->tl_mds[md_index].attr;
         if (*md_map_p & UCS_BIT(md_index)) {
             /* already registered, use previous memh */
+            ucs_assert((prev_memh_index < prev_num_memh) &&
+                       (prev_memh_index < prev_num_memh_saved));
```

Comment:
why also need to compare to prev_num_memh_saved? not enough to compare to prev_num_memh?

Reply:
clang can't understand that `prev_memh_index < prev_num_memh` is enough